landlab/components/lateral_erosion/lateral_erosion.py

```diff
@@ -0,0 +1,631 @@
+                        petlat = -Kl[i] * da[i] * max_slopes[i] * inv_rad_curv
+                        # the calculated potential lateral erosion is mutiplied by the length of the node
+                        # and the bank height, then added to an array, vol_lat_dt, for volume eroded
+                        # laterally  *per timestep* at each node. This vol_lat_dt is reset to zero for
+                        # each timestep loop. vol_lat_dt is added to itself in case more than one primary
+                        # nodes are laterally eroding this lat_node
+                        # volume of lateral erosion per timestep
+                        vol_lat_dt[lat_node] += abs(petlat) * grid.dx * wd
+
+            # send sediment downstream. sediment eroded from vertical incision
+            # and lateral erosion is sent downstream
+            qs_in[flowdirs[i]] += (
+                qs_in[i] - (dzver[i] * grid.dx ** 2) - (petlat * grid.dx * wd)
+            )  # qsin to next node
+        qs[:] = qs_in - (dzver * grid.dx ** 2)
+        dzdt[:] = dzver * dt
+        vol_lat[:] += vol_lat_dt * dt
+        # this loop determines if enough lateral erosion has happened to change
+        # the height of the neighbor node.
+        for i in dwnst_nodes:
+            lat_node = lat_nodes[i]
+            wd = 0.4 * (da[i] * runoffms) ** 0.35
+            if lat_node > 0:  # greater than zero now bc inactive neighbors are value -1
+                if z[lat_node] > z[i]:
+                    # vol_diff is the volume that must be eroded from lat_node so that its
+                    # elevation is the same as node downstream of primary node
+                    # UC model: this would represent undercutting (the water height at
+                    # node i), slumping, and instant removal.
+                    if UC == 1:
+                        voldiff = (z[i] + wd - z[flowdirs[i]]) * grid.dx ** 2
+                    # TB model: entire lat node must be eroded before lateral erosion
+                    # occurs
+                    if TB == 1:
+                        voldiff = (z[lat_node] - z[flowdirs[i]]) * grid.dx ** 2
+                    # if the total volume eroded from lat_node is greater than the volume
+                    # needed to be removed to make node equal elevation,
+                    # then instantaneously remove this height from lat node. already has
+                    # timestep in it
+                    if vol_lat[lat_node] >= voldiff:
+                        dzlat[lat_node] = z[flowdirs[i]] - z[lat_node]  # -0.001
+                        # after the lateral node is eroded, reset its volume eroded to
+                        # zero
+                        vol_lat[lat_node] = 0.0
+        # combine vertical and lateral erosion
+        dz = dzdt + dzlat
+        # change height of landscape
+        z[:] = dz + z
+        grid.at_node["topographic__elevation"][grid.core_nodes] = z[grid.core_nodes]
```

Comment:
I'm pretty sure that this line (and its equivalents in the adaptive function) is not necessary because python uses pointers, you've effectively set the values for z on the grid, when you write 

```python
z[;] = dz + z
```

I'd also recommend writing

```python
z[:] += dz
```

instead.

Reply:
removed the unnecessary lines.